tests/extmod/machine_disable_irq.py

```diff
@@ -0,0 +1,26 @@
+# Test executing Python code while IRQs disabled.
+try:
+    from machine import disable_irq, enable_irq
+    from time import ticks_us, ticks_diff
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+
+# Structured to also nest disable_irq() calls
+def f(n):
+    st_irq = disable_irq()
+    if n:
+        f(n - 1)
+    else:
+        # busy-wait in a tight loop for 1ms, to simulate doing some work in a critical section
+        t0 = ticks_us()
+        while ticks_diff(ticks_us(), t0) < 1000:
```

Comment:
I suggest just making this a fixed-iteration loop calling `ticks_us()` each time (so it actually does some work).  Eg:
```py
for _ in range(100):
    ticks_us()
```
That should still trigger the original bug because there are more than 32 iterations, so the VM will try to release the GIL.

Then separately (one day) we could add a specific test for using `ticks_us()` while interrupts are disabled, and specify which ports that actually works on, and maybe fix other ports.

Reply:
Argh, of course!

Have implemented the suggestion, confirmed that the test still fails on ESP32-C3 on master and passes with the fix. Pyboard V1.1 also pases this test now.

Reply:
Thanks for updating.  I also tested on PYBD-SF2 and it works now.